### PR TITLE
Add Github APP token task

### PR DESCRIPTION
**Pull request by chmouel**
This add the github app token, this would help to get a user token from a github
app which can be perused to do github operations.

Flow for github apps to work on tekton can usually going like this :

* github events
       -> triggers event listenner 
       -> start a pipeline
       ->  -> get user token from app (with this task)
       ->  -> do operation (like ci checks or others) on github with perusing
              the token from the previous task

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Complies with [Catalog Orgainization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ X Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/github-app-token/0.1/samples/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: github-app-secret
+type: Opaque
+stringData:
+  private.key: |-
+   -----BEGIN RSA PRIVATE KEY-----
+   MIICXAIBAAKBgQCxiQEUg+S3LmCUDVUaa6q1GM5+phvoL4UXkzr8aa1VZGJCWEEQ
```
Comment (chmouel) [inline task/github-app-token/0.1/samples/secret.yaml:9]: Thi is a random private key generated from https://travistidwell.com/jsencrypt/demo/ not an actual one (i had some git bots bugging me i have commited a rsa key)

**Comment by chmouel** [comment]
new force-push : 

* Made the docker images way smaller (from 900 to 80mb)
* Use results variable

File: task/github-app-token/0.1/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+A task to get a user token from a github application
+
+## Workspaces
+
+- **secrets**: A workspace containing the private key of the application.
+
+## Secret
+
+This GitHUB applications needs a private key to sign your request with JWT.
```
Comment (piyush-garg) [inline task/github-app-token/0.1/README.md:11]: nit GitHUB -> GitHub

File: task/github-app-token/0.1/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+### Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/task/github-app-token/0.1/github-app-token.yaml
+```
+
+
+## Usage
+
+After creating the task with the parameters, you should have the token results in the task which can
```
Comment (piyush-garg) [inline task/github-app-token/0.1/README.md:36]: nit `token as results`

File: task/github-app-token/0.1/github-app-token.yaml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  name: github-app-token
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: github
+    tekton.dev/displayName: "github app token"
+spec:
+  description: >-
+    Retrive a usee token from a GitHUB application
```
Comment (piyush-garg) [inline task/github-app-token/0.1/github-app-token.yaml:13]: nit: `user` and `GitHub`

**Review by piyush-garg — APPROVED** [review]
/lgtm

minor nits

File: task/github-app-token/0.1/github-app-token.yaml [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+    This task will get a user token for an installation_id for a GitHUB application.
```
Comment (vinamra28) [inline task/github-app-token/0.1/github-app-token.yaml:15]:
```suggestion
    This task will get a user token for an installation_id for a GitHub application.
```

**Review by vinamra28 — COMMENTED** [review]
just a few nits otherwise looks good :smile:

**Review by Divyansh42 — APPROVED** [review]
/lgtm

**Review by vdemeester — APPROVED** [review]
/meow

**Pull request merged**